tool/planet/start.go

```diff
@@ -349,6 +356,47 @@ func addCloudOptions(c *Config) error {
 	return nil
 }
 
+// configureProxy separates http_proxy environment variables into their own config file, and appends no_proxy
+// directives that may be missing in customer provided configuration
+func configureProxy(c *Config) {
+	// Attempt to find any current proxy settings in the variable names golang supports
+	// https://github.com/golang/net/blob/c21de06aaf072cea07f3a65d6970e5c7d8b6cd6d/http/httpproxy/proxy.go#L91-L98
+	proxy := []string{
+		"HTTP_PROXY",
+		"http_proxy",
+		"HTTPS_PROXY",
+		"https_proxy",
+	}
+
+	noProxy := map[string]string{
+		"NO_PROXY": c.Env.Delete("NO_PROXY"),
+		"no_proxy": c.Env.Delete("no_proxy"),
+	}
+
+	found := false
+	for k, v := range noProxy {
+		if len(v) != 0 {
+			// TODO(knisbet) we should see if there is a way to not NO_PROXY every IP address
+			// but it's difficult because we need to know each of the nodes IP addresses, which can be added
+			// after the cluster starts. Alternatively we would need to make internal connections to <ip>.ip.local and
+			// have coredns convert the IP addresses for us as a DNS query.
+			c.ProxyEnv.Upsert(k, strings.Join([]string{v, "0.0.0.0/0", ".local"}, ","))
+			found = true
+		}
+	}
+
+	// Move proxy settings from regular Environment to Proxy environment settings so not all planet processes
+	// try and use the http_proxy.
+	for _, k := range proxy {
+		c.ProxyEnv.Upsert(k, c.Env.Delete(k))
+	}
+
+	// If we're unable to locate a NO_PROXY config, create default settings
+	if !found {
+		c.ProxyEnv.Upsert("NO_PROXY", strings.Join([]string{"0.0.0.0/0", ".local"}, ","))
```

Comment:
Maybe move `[]string{"0.0.0.0/0", ".local"}` into a some kind of constant (even if in this module) as it's being used in a few places?